doorman/rules.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+from collections import namedtuple
+
+from doorman.models import Rule
+from doorman.utils import extract_results
+
+
+# Note: the fields are:
+#   - rule_id               The ID of the Rule that created this match
+#   - node                  The node that this Rule match applies to
+#   - action                The action that took place ('added', 'removed', or None)
+#   - match                 Some additional details about this match (any type)
+RuleMatch = namedtuple('RuleMatch', ['rule_id', 'node', 'action', 'match'])
+
+
+class BaseRule(object):
+    """
+    BaseRule is the base class for all rules in Doorman. It defines the
+    interface that should be implemented by classes that know how to take
+    incoming log data and determine if a rule has been triggered.
+    """
+    def __init__(self, rule_id, action, config):
+        self.rule_id = rule_id
+        self.node_name = config.get('node_name')
+
+    def handle_log_entry(self, entry, node):
+        """
+        This function processes an incoming log entry.  It normalizes the data,
+        validates the node name (if that filter was given), and then dispatches
+        to the underlying rule.
+        """
+        if self.node_name is not None and node['host_identifier'] != self.node_name:
```

Comment:
I think we should pass the entire JSON object along to this method. The request object contains the `node_key`, which would give us the opportunity to lookup the node in the database and obtains it's `hostIdentifier` if we wanted to. Also, in the event format, a `hostIdentifier` is included in each log result, giving us another opportunity to filter there, or report changes.

See the [batch](https://github.com/mwielgoszewski/doorman/blob/b6e056b/tests/test_functional.py#L316) vs [event](https://github.com/mwielgoszewski/doorman/blob/b6e056b/tests/test_functional.py#L374) test cases for some examples.